This seems to fix the following build issue of qosify on OpenWrt 24.10 for me on Debian Sid with ccache installed, a PATH that contains "/usr/lib/ccache" and `CONFIG_BPF_TOOLCHAIN_HOST=y`:

```
$ make package/network/config/qosify/{clean,compile} -j1 V=s
WARNING: Makefile 'package/utils/audit/Makefile' has a dependency on 'libev', which does not exist
WARNING: Makefile 'package/utils/busybox/Makefile' has a dependency on 'libpam', which does not exist
WARNING: Makefile 'package/utils/busybox/Makefile' has a dependency on 'libpam', which does not exist
WARNING: Makefile 'package/utils/busybox/Makefile' has a build dependency on 'libpam', which does not exist
WARNING: Makefile 'package/boot/kexec-tools/Makefile' has a dependency on 'liblzma', which does not exist
WARNING: Makefile 'package/network/services/lldpd/Makefile' has a dependency on 'libnetsnmp', which does not exist
WARNING: Makefile 'package/utils/policycoreutils/Makefile' has a dependency on 'libpam', which does not exist
WARNING: Makefile 'package/utils/policycoreutils/Makefile' has a dependency on 'libpam', which does not exist
WARNING: Makefile 'package/utils/policycoreutils/Makefile' has a build dependency on 'libpam', which does not exist
make[2]: Entering directory '/home/linus/dev-priv/openwrt/openwrt-24.10/scripts/config'
make[2]: 'conf' is up to date.
make[2]: Leaving directory '/home/linus/dev-priv/openwrt/openwrt-24.10/scripts/config'
make[1]: Entering directory '/home/linus/dev-priv/openwrt/openwrt-24.10'
make[2]: Entering directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/network/config/qosify'
rm -rf /home/linus/dev-priv/openwrt/openwrt-24.10/build_dir/target-x86_64_musl/linux-x86_64/qosify-2024.09.20~1501e093
/home/linus/dev-priv/openwrt/openwrt-24.10/scripts/ipkg-remove qosify /home/linus/dev-priv/openwrt/openwrt-24.10/bin/targets/x86/64/packages/qosify_2024.09.20~1501e093-r1_x86_64.ipk
removed '/home/linus/dev-priv/openwrt/openwrt-24.10/bin/targets/x86/64/packages/qosify_2024.09.20~1501e093-r1_x86_64.ipk'
rm -f /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/stamp/.qosify_installed
rm -f /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/packages/qosify.list
make[2]: Leaving directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/network/config/qosify'
time: package/network/config/qosify/clean#0.23#0.36#0.71
make[1]: Leaving directory '/home/linus/dev-priv/openwrt/openwrt-24.10'
WARNING: Makefile 'package/utils/audit/Makefile' has a dependency on 'libev', which does not exist
WARNING: Makefile 'package/utils/busybox/Makefile' has a dependency on 'libpam', which does not exist
WARNING: Makefile 'package/utils/busybox/Makefile' has a dependency on 'libpam', which does not exist
WARNING: Makefile 'package/utils/busybox/Makefile' has a build dependency on 'libpam', which does not exist
WARNING: Makefile 'package/boot/kexec-tools/Makefile' has a dependency on 'liblzma', which does not exist
WARNING: Makefile 'package/network/services/lldpd/Makefile' has a dependency on 'libnetsnmp', which does not exist
WARNING: Makefile 'package/utils/policycoreutils/Makefile' has a dependency on 'libpam', which does not exist
WARNING: Makefile 'package/utils/policycoreutils/Makefile' has a dependency on 'libpam', which does not exist
WARNING: Makefile 'package/utils/policycoreutils/Makefile' has a build dependency on 'libpam', which does not exist
make[2]: Entering directory '/home/linus/dev-priv/openwrt/openwrt-24.10/scripts/config'
make[2]: 'conf' is up to date.
make[2]: Leaving directory '/home/linus/dev-priv/openwrt/openwrt-24.10/scripts/config'
make[1]: Entering directory '/home/linus/dev-priv/openwrt/openwrt-24.10'
+ mkdir -p /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl
+ cd /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl
+ mkdir -p bin lib stamp usr/include usr/lib
mkdir -p /home/linus/dev-priv/openwrt/openwrt-24.10/build_dir/target-x86_64_musl/stamp
touch /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/.prepared
make[2]: Entering directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/libs/toolchain'
echo "libc" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/toolchain.default.install
echo "libgcc" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/toolchain.default.install
echo "libpthread" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/toolchain.default.install
echo "librt" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/toolchain.default.install
make[2]: Leaving directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/libs/toolchain'
time: package/libs/toolchain/compile#0.21#0.10#0.29
make[2]: Entering directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/kernel/bpf-headers'
make[2]: Leaving directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/kernel/bpf-headers'
time: package/kernel/bpf-headers/compile#1.12#2.66#2.81
make[2]: Entering directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/firmware/linux-firmware'
echo "bnx2-firmware" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux-firmware.default.install
echo "i915-firmware-dmc" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux-firmware.default.install
echo "r8169-firmware" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux-firmware.default.install
make[2]: Leaving directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/firmware/linux-firmware'
time: package/firmware/linux-firmware/compile#1.97#0.70#2.62
make[2]: Entering directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/kernel/gpio-button-hotplug'
make[2]: Nothing to be done for 'compile'.
make[2]: Leaving directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/kernel/gpio-button-hotplug'
time: package/kernel/gpio-button-hotplug/compile#0.15#0.09#0.21
make[2]: Entering directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/kernel/linux'
echo "kernel" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-crypto-crc32c" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-crypto-hash" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-fs-vfat" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-hwmon-core" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-i2c-core" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-i2c-algo-bit" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-input-core" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-lib-crc-ccitt" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-lib-crc32c" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-libphy" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-phylink" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-mii" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-mdio-devres" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-phy-realtek" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-r8169" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-e1000" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-e1000e" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-igb" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-ixgbe" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-tg3" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-ifb" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-forcedeth" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-bnx2" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-pcs-xpcs" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-stmmac-core" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-igc" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-amazon-ena" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-nf-reject" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-nf-reject6" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-nf-conntrack" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-nf-conntrack6" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-nf-log" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-nf-log6" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-nf-nat" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-nf-flow" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-nfnetlink" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-nft-core" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-nft-nat" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-nft-offload" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-nft-fib" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-slhc" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-ppp" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-pppox" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-pppoe" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-sched-core" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-sched-bpf" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-sched-cake" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-mdio" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-nls-base" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-nls-cp437" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-nls-iso8859-1" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-nls-utf8" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-dma-buf" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-pps" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-ptp" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-acpi-video" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-backlight" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-fb" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-fb-cfb-fillrect" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-fb-cfb-copyarea" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-fb-cfb-imgblt" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-fb-sys-fops" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-fb-sys-ram" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-drm" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-drm-buddy" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-drm-display-helper" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
WARNING: kmod-drm-exec is not available in the kernel config - generating empty package
echo "kmod-drm-exec" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-drm-ttm" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-drm-ttm-helper" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-drm-kms-helper" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
WARNING: kmod-drm-suballoc-helper is not available in the kernel config - generating empty package
echo "kmod-drm-suballoc-helper" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-drm-i915" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-amd-xgbe" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
echo "kmod-dwmac-intel" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/linux.default.install
make[2]: Leaving directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/kernel/linux'
time: package/kernel/linux/compile#2.19#0.22#2.41
make[2]: Entering directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/libs/zlib'
echo "zlib" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/zlib.default.install
make[2]: Leaving directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/libs/zlib'
time: package/libs/zlib/compile#0.21#0.15#0.33
make[2]: Entering directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/utils/bzip2'
make[2]: Leaving directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/utils/bzip2'
time: package/utils/bzip2/compile#0.48#0.24#0.70
make[2]: Entering directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/libs/argp-standalone'
make[2]: Leaving directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/libs/argp-standalone'
time: package/libs/argp-standalone/compile#0.30#0.18#0.44
make[2]: Entering directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/libs/musl-fts'
make[2]: Leaving directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/libs/musl-fts'
time: package/libs/musl-fts/compile#0.33#0.20#0.50
make[2]: Entering directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/libs/elfutils'
echo "libelf" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/elfutils.default.install
make[2]: Leaving directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/libs/elfutils'
time: package/libs/elfutils/compile#0.23#0.26#0.45
make[2]: Entering directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/libs/libbpf'
echo "libbpf" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/libbpf.default.install
make[2]: Leaving directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/libs/libbpf'
time: package/libs/libbpf/compile#0.19#0.23#0.38
make[2]: Entering directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/libs/libnl-tiny'
echo "libnl-tiny" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/libnl-tiny.default.install
make[2]: Leaving directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/libs/libnl-tiny'
time: package/libs/libnl-tiny/compile#0.19#0.21#0.37
make[2]: Entering directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/libs/libjson-c'
echo "libjson-c" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/libjson-c.default.install
make[2]: Leaving directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/libs/libjson-c'
time: package/libs/libjson-c/compile#0.31#0.31#0.57
make[2]: Entering directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/utils/lua'
make[2]: Leaving directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/utils/lua'
time: package/utils/lua/compile#0.33#0.34#0.61
make[2]: Entering directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/libs/libubox'
echo "libubox" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/libubox.default.install
echo "libblobmsg-json" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/libubox.default.install
echo "jshn" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/libubox.default.install
echo "libjson-script" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/libubox.default.install
make[2]: Leaving directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/libs/libubox'
time: package/libs/libubox/compile#0.35#0.35#0.65
make[2]: Entering directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/system/ubus'
echo "libubus" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/ubus.default.install
echo "ubus" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/ubus.default.install
echo "ubusd" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/ubus.default.install
make[2]: Leaving directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/system/ubus'
time: package/system/ubus/compile#0.21#0.23#0.41
make[2]: Entering directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/libs/libmnl'
echo "libmnl" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/libmnl.default.install
make[2]: Leaving directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/libs/libmnl'
time: package/libs/libmnl/compile#0.40#0.21#0.57
make[2]: Entering directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/libs/libnftnl'
echo "libnftnl" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/libnftnl.default.install
make[2]: Leaving directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/libs/libnftnl'
time: package/libs/libnftnl/compile#0.21#0.21#0.37
make[2]: Entering directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/network/utils/iptables'
make[2]: Leaving directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/network/utils/iptables'
time: package/network/utils/iptables/compile#0.26#0.27#0.48
make[2]: Entering directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/network/utils/iproute2'
echo "tc-tiny" >> /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/pkginfo/iproute2.tctiny.install
make[2]: Leaving directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/network/utils/iproute2'
time: package/network/utils/iproute2/tctiny/compile#0.30#0.28#0.53
make[2]: Entering directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/network/config/qosify'
touch /home/linus/dev-priv/openwrt/openwrt-24.10/build_dir/target-x86_64_musl/linux-x86_64/qosify-2024.09.20~1501e093/.prepared_64c4e68b91b1ade235fc12691a40e41a_6664517399ebbbc92a37c5bb081b5c53_check
. /home/linus/dev-priv/openwrt/openwrt-24.10/include/shell.sh; zstdcat /home/linus/dev-priv/openwrt/openwrt-24.10/dl/qosify-2024.09.20~1501e093.tar.zst | tar -C /home/linus/dev-priv/openwrt/openwrt-24.10/build_dir/target-x86_64_musl/linux-x86_64/qosify-2024.09.20~1501e093/.. -xf -
[ ! -d ./src/ ] || cp -fpR ./src/. /home/linus/dev-priv/openwrt/openwrt-24.10/build_dir/target-x86_64_musl/linux-x86_64/qosify-2024.09.20~1501e093
touch /home/linus/dev-priv/openwrt/openwrt-24.10/build_dir/target-x86_64_musl/linux-x86_64/qosify-2024.09.20~1501e093/.prepared_64c4e68b91b1ade235fc12691a40e41a_6664517399ebbbc92a37c5bb081b5c53
rm -f /home/linus/dev-priv/openwrt/openwrt-24.10/build_dir/target-x86_64_musl/linux-x86_64/qosify-2024.09.20~1501e093/.configured_*
rm -f /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/stamp/.qosify_installed
mkdir -p /home/linus/dev-priv/openwrt/openwrt-24.10/build_dir/target-x86_64_musl/linux-x86_64/qosify-2024.09.20~1501e093
(cd /home/linus/dev-priv/openwrt/openwrt-24.10/build_dir/target-x86_64_musl/linux-x86_64/qosify-2024.09.20~1501e093; CFLAGS="-Os -pipe -fno-caller-saves -fno-plt -fhonour-copts -fmacro-prefix-map=/home/linus/dev-priv/openwrt/openwrt-24.10/build_dir/target-x86_64_musl/linux-x86_64/qosify-2024.09.20~1501e093=qosify-2024.09.20~1501e093 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro   -Wno-error=deprecated-declarations -I/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/usr/include/libnl-tiny -I/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/usr/include " CXXFLAGS="-Os -pipe -fno-caller-saves -fno-plt -fhonour-copts -fmacro-prefix-map=/home/linus/dev-priv/openwrt/openwrt-24.10/build_dir/target-x86_64_musl/linux-x86_64/qosify-2024.09.20~1501e093=qosify-2024.09.20~1501e093 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro   -Wno-error=deprecated-declarations -I/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/usr/include/libnl-tiny -I/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/usr/include " LDFLAGS="-L/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/toolchain-x86_64_gcc-13.3.0_musl/usr/lib -L/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/toolchain-x86_64_gcc-13.3.0_musl/lib -fuse-ld=bfd -znow -zrelro   " cmake --no-warn-unused-cli -DCMAKE_SYSTEM_NAME=Linux -DCMAKE_SYSTEM_VERSION=1 -DCMAKE_SYSTEM_PROCESSOR=x86_64 -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_FLAGS_RELEASE="-DNDEBUG" -DCMAKE_CXX_FLAGS_RELEASE="-DNDEBUG" -DCMAKE_C_COMPILER_LAUNCHER="" -DCMAKE_C_COMPILER="/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/toolchain-x86_64_gcc-13.3.0_musl/bin/x86_64-openwrt-linux-musl-gcc" -DCMAKE_CXX_COMPILER_LAUNCHER="" -DCMAKE_CXX_COMPILER="/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/toolchain-x86_64_gcc-13.3.0_musl/bin/x86_64-openwrt-linux-musl-g++" -DCMAKE_ASM_COMPILER_LAUNCHER="" -DCMAKE_ASM_COMPILER="/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/toolchain-x86_64_gcc-13.3.0_musl/bin/x86_64-openwrt-linux-musl-gcc" -DCMAKE_EXE_LINKER_FLAGS:STRING="-L/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/toolchain-x86_64_gcc-13.3.0_musl/usr/lib -L/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/toolchain-x86_64_gcc-13.3.0_musl/lib -fuse-ld=bfd -znow -zrelro  " -DCMAKE_MODULE_LINKER_FLAGS:STRING="-L/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/toolchain-x86_64_gcc-13.3.0_musl/usr/lib -L/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/toolchain-x86_64_gcc-13.3.0_musl/lib -fuse-ld=bfd -znow -zrelro   -Wl,-Bsymbolic-functions" -DCMAKE_SHARED_LINKER_FLAGS:STRING="-L/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/toolchain-x86_64_gcc-13.3.0_musl/usr/lib -L/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/toolchain-x86_64_gcc-13.3.0_musl/lib -fuse-ld=bfd -znow -zrelro   -Wl,-Bsymbolic-functions" -DCMAKE_AR="/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/toolchain-x86_64_gcc-13.3.0_musl/bin/x86_64-openwrt-linux-musl-gcc-ar" -DCMAKE_NM="/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/toolchain-x86_64_gcc-13.3.0_musl/bin/x86_64-openwrt-linux-musl-gcc-nm" -DCMAKE_RANLIB="/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/toolchain-x86_64_gcc-13.3.0_musl/bin/x86_64-openwrt-linux-musl-gcc-ranlib" -DCMAKE_FIND_ROOT_PATH="/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/usr;/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/toolchain-x86_64_gcc-13.3.0_musl" -DCMAKE_FIND_ROOT_PATH_MODE_PROGRAM=BOTH -DCMAKE_FIND_ROOT_PATH_MODE_LIBRARY=ONLY -DCMAKE_FIND_ROOT_PATH_MODE_INCLUDE=ONLY -DCMAKE_STRIP=: -DCMAKE_INSTALL_PREFIX=/usr -DDL_LIBRARY=/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl -DCMAKE_PREFIX_PATH=/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl -DCMAKE_SKIP_RPATH=TRUE -DCMAKE_EXPORT_PACKAGE_REGISTRY=FALSE -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=TRUE -DCMAKE_FIND_USE_PACKAGE_REGISTRY=FALSE -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=TRUE -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=FALSE -DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=TRUE -DCMAKE_GENERATOR="Ninja" -DLIBNL_LIBS=-lnl-tiny /home/linus/dev-priv/openwrt/openwrt-24.10/build_dir/target-x86_64_musl/linux-x86_64/qosify-2024.09.20~1501e093 )
Not searching for unused variables given on the command line.
-- The C compiler identification is GNU 13.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/toolchain-x86_64_gcc-13.3.0_musl/bin/x86_64-openwrt-linux-musl-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Configuring done (0.4s)
-- Generating done (0.0s)
-- Build files have been written to: /home/linus/dev-priv/openwrt/openwrt-24.10/build_dir/target-x86_64_musl/linux-x86_64/qosify-2024.09.20~1501e093
touch /home/linus/dev-priv/openwrt/openwrt-24.10/build_dir/target-x86_64_musl/linux-x86_64/qosify-2024.09.20~1501e093/.configured_a17fb5ef857664f03cd0ce37cc5ea591
rm -f /home/linus/dev-priv/openwrt/openwrt-24.10/build_dir/target-x86_64_musl/linux-x86_64/qosify-2024.09.20~1501e093/.built
touch /home/linus/dev-priv/openwrt/openwrt-24.10/build_dir/target-x86_64_musl/linux-x86_64/qosify-2024.09.20~1501e093/.built_check
/usr/lib/ccache/clang -g -target mips64el-linux-gnu -nostdinc -isystem /home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/toolchain-x86_64_gcc-13.3.0_musl/lib/gcc/*/*/include -isystem/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/toolchain-x86_64_gcc-13.3.0_musl/usr/include -isystem/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/toolchain-x86_64_gcc-13.3.0_musl/include -isystem/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/toolchain-x86_64_gcc-13.3.0_musl/include/fortify -I/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/bpf-headers/arch/mips/include -I/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/bpf-headers/arch/mips/include/asm/mach-generic -I/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/bpf-headers/arch/mips/include/generated -I/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/bpf-headers/include -I/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/bpf-headers/arch/mips/include/uapi -I/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/bpf-headers/arch/mips/include/generated/uapi -I/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/bpf-headers/include/uapi -I/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/bpf-headers/include/generated/uapi -I/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/bpf-headers/tools/lib -I/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/bpf-headers/tools/testing/selftests -I/home/linus/dev-priv/openwrt/openwrt-24.10/staging_dir/target-x86_64_musl/bpf-headers/samples/bpf -include linux/kconfig.h -include asm_goto_workaround.h -I/home/linus/dev-priv/openwrt/openwrt-24.10/build_dir/target-x86_64_musl/linux-x86_64/qosify-2024.09.20~1501e093 -D__KERNEL__ -D__BPF_TRACING__ -DCONFIG_GENERIC_CSUM -D__TARGET_ARCH_mips -mlittle-endian -fno-stack-protector -Wall -Wno-unused-value -Wno-pointer-sign -Wno-compare-distinct-pointer-types -Wno-gnu-variable-sized-type-not-at-end -Wno-address-of-packed-member -Wno-tautological-compare -Wno-unknown-warning-option -fno-asynchronous-unwind-tables -Wno-uninitialized -Wno-unused-variable -Wno-unused-label -O2 -emit-llvm -Xclang -disable-llvm-passes  -c /home/linus/dev-priv/openwrt/openwrt-24.10/build_dir/target-x86_64_musl/linux-x86_64/qosify-2024.09.20~1501e093/qosify-bpf.c -o /home/linus/dev-priv/openwrt/openwrt-24.10/build_dir/target-x86_64_musl/linux-x86_64/qosify-2024.09.20~1501e093/qosify-bpf.bc
/usr/lib/ccache//opt -O2 -mtriple=bpfel < /home/linus/dev-priv/openwrt/openwrt-24.10/build_dir/target-x86_64_musl/linux-x86_64/qosify-2024.09.20~1501e093/qosify-bpf.bc > /home/linus/dev-priv/openwrt/openwrt-24.10/build_dir/target-x86_64_musl/linux-x86_64/qosify-2024.09.20~1501e093/qosify-bpf.opt
bash: line 1: /usr/lib/ccache//opt: No such file or directory
make[2]: *** [Makefile:80: /home/linus/dev-priv/openwrt/openwrt-24.10/build_dir/target-x86_64_musl/linux-x86_64/qosify-2024.09.20~1501e093/.built] Error 127
make[2]: Leaving directory '/home/linus/dev-priv/openwrt/openwrt-24.10/package/network/config/qosify'
time: package/network/config/qosify/compile#0.45#0.66#1.14
    ERROR: package/network/config/qosify failed to build.
make[1]: *** [package/Makefile:179: package/network/config/qosify/compile] Error 1
make[1]: Leaving directory '/home/linus/dev-priv/openwrt/openwrt-24.10'
make: *** [/home/linus/dev-priv/openwrt/openwrt-24.10/include/toplevel.mk:233: package/network/config/qosify/compile] Error 2
nice make package/network/config/qosify/{clean,compile} -j1 V=s  24.10s user 12.56s system 106% cpu 34.402 total
```

Thanks for this patch!

--- a/include/bpf.mk
+++ b/include/bpf.mk
@@ -10,7 +10,7 @@ ifneq ($(CONFIG_USE_LLVM_HOST),)
   else
     BPF_PATH:=$(PATH)
   endif
-  CLANG:=$(firstword $(shell PATH='$(BPF_PATH)' command -v clang clang-13 clang-12 clang-11))
+  CLANG:=$(firstword $(shell PATH='$(BPF_PATH)' command -v clang clang-13 clang-12))
   LLVM_VER:=$(subst clang,,$(notdir $(CLANG)))
 endif
 ifneq ($(CONFIG_USE_LLVM_PREBUILT),)

--- a/include/bpf.mk
+++ b/include/bpf.mk
@@ -1,30 +1,40 @@
 BPF_DEPENDS := @HAS_BPF_TOOLCHAIN +@NEED_BPF_TOOLCHAIN
-LLVM_VER:=
 
 CLANG_MIN_VER:=12
 
 ifneq ($(CONFIG_USE_LLVM_HOST),)
+  find-llvm-tool=$(firstword $(shell PATH='$(BPF_PATH)' command -v $(1) || echo '$(firstword $(1))-not-found'))
+
   BPF_TOOLCHAIN_HOST_PATH:=$(call qstrip,$(CONFIG_BPF_TOOLCHAIN_HOST_PATH))
   ifneq ($(BPF_TOOLCHAIN_HOST_PATH),)
     BPF_PATH:=$(BPF_TOOLCHAIN_HOST_PATH)/bin:$(PATH)
   else
     BPF_PATH:=$(PATH)
   endif
-  CLANG:=$(firstword $(shell PATH='$(BPF_PATH)' command -v clang clang-13 clang-12))
+  CLANG:=$(call find-llvm-tool,clang clang-13 clang-12)
   LLVM_VER:=$(subst clang,,$(notdir $(CLANG)))
-endif
-ifneq ($(CONFIG_USE_LLVM_PREBUILT),)
-  CLANG:=$(TOPDIR)/llvm-bpf/bin/clang
-endif
-ifneq ($(CONFIG_USE_LLVM_BUILD),)
-  CLANG:=$(STAGING_DIR_HOST)/llvm-bpf/bin/clang
-endif
 
-LLVM_PATH:=$(dir $(CLANG))
-LLVM_LLC:=$(LLVM_PATH)/llc$(LLVM_VER)
-LLVM_DIS:=$(LLVM_PATH)/llvm-dis$(LLVM_VER)
-LLVM_OPT:=$(LLVM_PATH)/opt$(LLVM_VER)
-LLVM_STRIP:=$(LLVM_PATH)/llvm-strip$(LLVM_VER)
+  BPF_PATH:=$(dir $(CLANG)):$(BPF_PATH)
+  LLVM_LLC:=$(call find-llvm-tool,llc$(LLVM_VER))
+  LLVM_DIS:=$(call find-llvm-tool,llvm-dis$(LLVM_VER))
+  LLVM_OPT:=$(call find-llvm-tool,opt$(LLVM_VER))
+  LLVM_STRIP:=$(call find-llvm-tool,llvm-strip$(LLVM_VER))
+else
+  LLVM_PATH:=/invalid
+
+  ifneq ($(CONFIG_USE_LLVM_PREBUILT),)
+    LLVM_PATH:=$(TOPDIR)/llvm-bpf/bin
+  endif
+  ifneq ($(CONFIG_USE_LLVM_BUILD),)
+    LLVM_PATH:=$(STAGING_DIR_HOST)/llvm-bpf/bin
+  endif
+
+  CLANG:=$(LLVM_PATH)/clang
+  LLVM_LLC:=$(LLVM_PATH)/llc
+  LLVM_DIS:=$(LLVM_PATH)/llvm-dis
+  LLVM_OPT:=$(LLVM_PATH)/opt
+  LLVM_STRIP:=$(LLVM_PATH)/llvm-strip
+endif
 
 BPF_KARCH:=mips
 BPF_ARCH:=mips$(if $(CONFIG_ARCH_64BIT),64)$(if $(CONFIG_BIG_ENDIAN),,el)

